src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+
+    private final PartitionCrossingChecker partitionCrossingChecker;
+
+    private final Predictor predictor;
+    private final Map<String, Integer> evidenceTypeMap;
+    private final double coverage;
+    private final double thresholdProbability;
+    private final Map<String, String> readGroupToLibraryMap;
+    private final Map<String, LibraryStatistics> libraryStatisticsMap;
+
+    private final EvidenceOverlapChecker evidenceOverlapChecker;
+    private final Map<BreakpointEvidence, int[]> rawFeatureCache;
+
+    private Iterator<SVIntervalTree.Entry<List<BreakpointEvidence>>> treeItr;
+    private Iterator<BreakpointEvidence> listItr;
+    private final SVIntervalTree<Double> genomeGaps;
+    private final SVIntervalTree<Double> umapS100Mappability;
+
+    public XGBoostEvidenceFilter(
+            final Iterator<BreakpointEvidence> evidenceItr,
+            final ReadMetadata readMetadata,
+            final StructuralVariationDiscoveryArgumentCollection.FindBreakpointEvidenceSparkArgumentCollection params,
+            final PartitionCrossingChecker partitionCrossingChecker
+    ) {
+        this.predictor = loadPredictor(params.svEvidenceFilterModelFile);
+        this.evidenceTypeMap = loadEvidenceTypeMap(params.svCategoricalVariablesFile);
+        this.genomeGaps = new SVIntervalTree<>();
+        loadGenomeIntervals(this.genomeGaps, params.svGenomeGapsFile, readMetadata);
+        loadGenomeIntervals(this.genomeGaps, params.svGenomeCentromeresFile, readMetadata);
+        this.umapS100Mappability = new SVIntervalTree<>();
+        loadGenomeIntervals(this.umapS100Mappability, params.svGenomeUmapS100File, readMetadata);
+
+        this.partitionCrossingChecker = partitionCrossingChecker;
+        this.coverage = readMetadata.getCoverage();
+        this.thresholdProbability = params.svEvidenceFilterThresholdProbability;
+        this.readGroupToLibraryMap = readMetadata.getReadGroupToLibraryMap();
+        this.libraryStatisticsMap = readMetadata.getAllLibraryStatistics();
+
+        this.evidenceOverlapChecker = new EvidenceOverlapChecker(evidenceItr, readMetadata, params.minEvidenceMapQ);
+        this.rawFeatureCache = new HashMap<>();
+
+        this.listItr = null;
+        this.treeItr = evidenceOverlapChecker.getTreeItr();
+    }
+
+
+    public static Predictor loadPredictor(final String modelFileLocation) {
+        ObjFunction.useFastMathExp(useFastMathExp);
+        try(final InputStream inputStream = getInputStream(modelFileLocation)) {
+            return new Predictor(inputStream);
+        } catch(Exception e) {
+            throw new GATKException(
+                    "Unable to load predictor from classifier file " + modelFileLocation + ": " + e.getMessage()
+            );
+        }
+    }
+
+    @VisibleForTesting
+    static InputStream getInputStream(final String fileLocation) throws IOException {
+        if(fileLocation.startsWith("gatk-resources::")) {
```

Comment:
I made that change. modelFile is optional, null by default. When null, the hard-coded resource file is loaded, otherwise BucketUtils loads the supplied file. Our hg19 and hg38 resource directories each have a copy of the classifier now. I'll have to do some investigation to see how well the hg38-trained classifier performs on hg19-aligned data.